components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/APIPersistence.java

```diff
@@ -144,6 +144,14 @@ void deleteAPIRevision(Organization org, String apiUUID, String revisionUUID, in
      */
     void deleteAPI(Organization org, String apiId) throws APIPersistenceException;
 
+    /**
+     * Delete all APIs
+     *
+     * @param org   Organization the API is owned by
+     * @throws APIPersistenceException
+     */
+    void deleteAllAPIs(Organization org) throws APIPersistenceException;
```

Comment:
This is not only about APIs right.. Even thought we consider about APIs only now, objective of this method is to delete anything that belongs to the organization? 
```suggestion
    void purge(Organization org) throws APIPersistenceException;
```

Reply:
Please update the discussion points we had @HiranyaKavishani

Reply:
Yes. But as per the organization deletion use case, the deletion of the MongoDB collection is done under the APIPurge functionality.So a better name should be similar to `deleteAllAPI`